storm-client/src/jvm/org/apache/storm/windowing/WaterMarkEventGenerator.java

```diff
@@ -45,17 +46,25 @@
     private final Map<GlobalStreamId, Long> streamToTs;
     private final ScheduledExecutorService executorService;
     private final int interval;
+    private final AtomicLong lastWaterMarkTs;
```

Comment:
It doesn't make much difference except the extra overhead of the AtomicLong object and we don't require the additional atomic operations from AtomicLong.